src/fondant/component_spec.py

```diff
@@ -33,15 +33,13 @@ def kubeflow2python_type(type_: str) -> t.Any:
     return lambda value: map_fn(value) if value != "None" else None  # type: ignore
 
 
-# TODO: Change after upgrading to kfp v2
-# :https://www.kubeflow.org/docs/components/pipelines/v2/data-types/parameters/
 python2kubeflow_type = {
-    "str": "String",
-    "int": "Integer",
-    "float": "Float",
-    "bool": "Boolean",
-    "dict": "JsonObject",
-    "list": "JsonArray",
+    "str": "STRING",
```

Comment:
Perhaps we can import this dict and use it? 

https://github.com/kubeflow/pipelines/blob/2b05ec867fad84e24fe73ef7515e3b5849297e79/sdk/python/kfp/dsl/types/type_utils.py#L56

We can maybe do the same for V1 as well instead of defining our own custom one 
https://github.com/kubeflow/pipelines/blob/2b05ec867fad84e24fe73ef7515e3b5849297e79/sdk/python/kfp/dsl/types/type_utils.py#L86

FYI I just stumbled on this when trying to look out for the v2 types since I couldn't find them in the docs

Reply:
I'll add it as a task